go/vt/vtgate/engine/route.go

```diff
@@ -591,3 +594,31 @@ func getQueries(query string, bvs []map[string]*querypb.BindVariable) []*querypb
 func orderByToString(in any) string {
 	return in.(OrderByParams).String()
 }
+
+func (route *Route) executeWarmingReplicaRead(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, queries []*querypb.BoundQuery) {
+	switch route.Opcode {
+	case Unsharded, Scatter, Equal, EqualUnique, IN:
+		// no-op
+	default:
+		return
+	}
+
+	if vcursor.GetWarmingReadsPercent() == 0 || rand.Intn(100) > vcursor.GetWarmingReadsPercent() {
+		return
+	}
+
+	replicaVCursor, ok := vcursor.CloneForReplicaWarming(ctx).(VCursor)
+
+	if !ok {
+		return
+	}
+
+	go func(replicaVCursor VCursor) {
+		rss, _, err := route.findRoute(ctx, replicaVCursor, bindVars)
+		if err != nil {
+			return
+		}
+
+		_, _ = replicaVCursor.ExecuteMultiShard(ctx, route, rss, queries, false /* rollbackOnError */, false /* autocommit */)
```

Comment:
Might be good to add some stats to add visibility into this.

Reply:
Ah I see in the main conversation now that there's talk about getting this at the tablet level. FWIW I think it could be useful here in case for whatever reason the queries fail to reach the tablets.